errbot/backends/irc.py

```diff
@@ -299,6 +313,7 @@ def __init__(self,
             self.send_public_message = rate_limited(channel_rate)(self.send_public_message)
         self._reconnect_on_kick = reconnect_on_kick
         self._pending_transfers = {}
+        self.__recently_joined_to = set()
```

Comment:
A single underscore denotes a "private" variable that should only be used internally by the class itself (note that in python, this is by convention only, it's not actually enforced).

Using a double underscore invokes something that is called name wrangling, which internally rewrites the variable name to have the classname prefixed. You use this if you want to be sure that if your class is subclassed and defines the same name, the two don't clash.

See https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references for the official explanation.

In this case, you do not want name wrangling to take place. It would in fact break subclasses from properly working. Hence, you should use a single underscore (the cases where you should use a double underscore are extremely rare in practice).